tools/ci_build/build.py

```diff
@@ -1487,24 +1487,27 @@ def generate_build_tree(
     cflags = None
     cxxflags = None
     ldflags = None
-    cudaflags = []
+    cudaflags = None
```

Comment:
maybe these flags lists should be moved inside the for loop? they aren't used outside and some of them just get reset in the loop.

Reply:
Updated.

Reply:
also, why use both None and an empty list `[]`? would it be simpler to just initialize everything as `[]`?

Reply:
Because cxxflags is copied from cflags. If I change all of the initializers  to "[]", I will also need to change some of the logic below.